src/XGBoostSharp.Tests/XGBClassifierTest.cs

```diff
@@ -128,6 +130,45 @@ public void XGBClassifierTest_UsingDMatrixDirectly()
         TestUtils.AssertAreEqual(expected, actual);
     }
 
+    [TestMethod]
```

Comment:
@madskonradsen I think we should refactor this test by adding a common utils method for aggregating the contributions. This can be shared between the classifier and regressor tests. This will also enable using the standard assert methods which takes into account delta on float etc.  See below:

```csharp
    [TestMethod]
    public void XGBClassifierTest_PredictionContributions()
    {
        var dataTrain = TestUtils.DataTrain;
        var labelsTrain = TestUtils.LabelsTrain;
        var dataTest = TestUtils.DataTest;

        using var sut = CreateSut();
        using var dMatrixTrain = new DMatrix(dataTrain, labelsTrain);
        sut.Fit(dMatrixTrain);

        using var dMatrixTest = new DMatrix(dataTest);
        var actualContributions = sut.Predict(dMatrixTest, predContribs: true);
        var predictedProbabilities = sut.PredictProbability(dMatrixTest);

        static float AggregateProbability(float[] values)
        {
            // In a binary classification problem, the raw margin is transformed
            // into a probability using the logistic function.
            return 1f / (1f + (float)Math.Exp(-values.Sum()));
        }

        // We want to check that the sum of the contributions(contributions +
        // last column, which is bias) is equal to the prediction.
        var actualProbabilities = TestUtils.AggregateContributions(actualContributions,
            AggregateProbability);
        var expectedProbabilities = predictedProbabilities.Select(x => x[1]).ToArray();

        TestUtils.AssertAreEqual(expectedProbabilities, actualProbabilities);
    }
```

New method on `TestUtils.cs`. This can also be reused in the XGBRegressorTest by injecting another aggregate method.
```csharp
    public static float[] AggregateContributions(Array actualContributions,
        Func<float[], float> aggregate)
    {
        // Dimensions of an Array are named Rank in C#.
        if (actualContributions.Rank == 2)
        {
            var aggregatedContributions = new float[actualContributions.GetLength(0)];
            // We want to check that the sum of the contributions(contributions
            // + last column, which is bias) is equal to the prediction.
            for (var i = 0; i < actualContributions.GetLength(0); i++)
            {
                var currentContribution = new float[actualContributions.GetLength(1)];
                for (var j = 0; j < actualContributions.GetLength(1); j++)
                {
                    currentContribution[j] = (float)actualContributions.GetValue(i, j);
                }
                var aggregatedContribution = aggregate(currentContribution);
                aggregatedContributions[i] = aggregatedContribution;
            }
            return aggregatedContributions;
        }
        else
        {
            throw new NotSupportedException("Only 2D arrays are supported.");
        }
    }
```

Reply:
You can reuse the method and do the same for the `XGBRegressorTest_PredictionContributions` :)